src/compiler/checker.ts

```diff
@@ -18542,6 +18560,11 @@ namespace ts {
                 }
                 return Ternary.False;
 
+                function countMessageChainBreadth(info: DiagnosticMessageChain[] | undefined): number {

```

Comment:
This is calculating the entire size of the message tree, right? (1) Seems a little expensive, though probably not excessively (2) seems like the name should talk about tree size.

(1) might not be true if info.length is usually 1 or chain.next is usually undefined.

Reply:
It's _usually_ linear, but overload compat errors can technically have multiple elements. It's not expensive in the grand scheme of things, really (especially since the depth is essentially capped by our relationship depth limit). Technically there could be circularly referential message chains (the type system allows it, anyway), which'd crash here, but if those ever get created we have a lot of problems.

Reply:
> Technically there could be circularly referential message chains (the type system allows it, anyway), which'd crash here, but if those ever get created we have a lot of problems.

You're kinda shooting yourself in the foot at that point. 😂

At the very least we'll overflow the stack if that ever happened here.